### PR TITLE
Add 1.19's support and support Minecraft version without patch version (1.18, 1.19)

### PR DESCRIPTION
**Pull request by TheoPierne**
- Adding 1.19 support.
- Support Minecraft version without patch version like 1.18, 1.19 to avoiding : https://github.com/dscalzi/Nebula/blob/8c224e3e398d76a9514e0975e7bd04506c30141d/src/util/MinecraftVersion.ts#L16

### CODE REVIEW

**Comment by dscalzi** [comment]
As I've said before, the problem is not supporting these new versions, the java handling in the main launcher has to be updated.

File: src/util/MinecraftVersion.ts [hunk 1]
```diff
@@ -23,8 +23,8 @@ export class MinecraftVersion {
 
     public getMajor(): number { return this.major }
     public getMinor(): number { return this.minor }
-    public getRevision(): number { return this.revision }
+    public getRevision(): number | null { return this.revision }
 
-    public toString(): string { return `${this.major}.${this.minor}.${this.revision}`}
+    public toString(): string { return this.revision === null ? `${this.major}.${this.minor}` : `${this.major}.${this.minor}.${this.revision}`}
```
Comment (GeekCornerGH) [inline src/util/MinecraftVersion.ts:28]: Seems good

File: src/resolver/forge/adapter/ForgeGradle3.resolver.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export class ForgeGradle3Adapter extends ForgeResolver {
         if(version.getMinor() === 12 && VersionUtil.isOneDotTwelveFG2(libraryVersion)) {
             return false
         }
-        return VersionUtil.isVersionAcceptable(version, [12, 13, 14, 15, 16])
+        return VersionUtil.isVersionAcceptable(version, [12, 13, 14, 15, 16, 17, 18, 19])
```
Comment (GeekCornerGH) [inline src/resolver/forge/adapter/ForgeGradle3.resolver.ts:34]:
```suggestion
        return VersionUtil.isVersionAcceptable(version, [12, 13, 14, 15, 16])
```
Adding other versions would be better in another PR

File: src/resolver/forge/adapter/ForgeGradle3.resolver.ts [hunk 3]
```diff
@@ -51,20 +51,13 @@ export class ForgeGradle3Adapter extends ForgeResolver {
     }
 
     private configure(): void {
-        // Configure for 13, 14, 15, 16
-        if(VersionUtil.isVersionAcceptable(this.minecraftVersion, [13, 14, 15, 16])) {
+        // Configure for 13, 14, 15, 16, 17, 18, 19
+        if(VersionUtil.isVersionAcceptable(this.minecraftVersion, [13, 14, 15, 16, 17, 18, 19])) {
```
Comment (GeekCornerGH) [inline src/resolver/forge/adapter/ForgeGradle3.resolver.ts:55]:
```suggestion
        // Configure for 13, 14, 15, 16
        if(VersionUtil.isVersionAcceptable(this.minecraftVersion, [13, 14, 15, 16])) {
```
Here again too

File: src/structure/spec_model/module/forgemod/ForgeMod113.struct.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ export class ForgeModStructure113 extends BaseForgeModStructure {
 
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     public static isForVersion(version: MinecraftVersion, libraryVersion: string): boolean {
-        return VersionUtil.isVersionAcceptable(version, [13, 14, 15, 16])
+        return VersionUtil.isVersionAcceptable(version, [13, 14, 15, 16, 17, 18, 19])
```
Comment (GeekCornerGH) [inline src/structure/spec_model/module/forgemod/ForgeMod113.struct.ts:16]:
```suggestion
        return VersionUtil.isVersionAcceptable(version, [13, 14, 15, 16])
```
Again, in a separate PR once 1.17+ has landed in Helios

**Review by GeekCornerGH — CHANGES_REQUESTED** [review]

**Comment by dscalzi** [comment]
Thanks, will be handled by #50

**Pull request closed without merging**